components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -534,6 +534,66 @@ void AtmosphereDriver::create_fields()
     m_field_mgrs[grid->name()]->registration_begins();
   }
 
+  // Before registering fields, check that Field Requests for tracers are compatible
+  {
```

Comment:
I wonder if this snippet should be moved somewhere, like in share/util, or maybe as a free function in `field_request.hpp`, something like

```c++
// ensure all requests are on the same page on whether the field belongs to $group_name
bool compatible_groups (std::vector<FieldRequest>& reqs, std::string& group_name);
```

(maybe with a better name)